src/pages/Debug/Report/DebugReportPage.tsx

```diff
@@ -1,35 +1,107 @@
 import type {StackScreenProps} from '@react-navigation/stack';
-import React from 'react';
+import React, {useMemo} from 'react';
 import {View} from 'react-native';
 import {useOnyx} from 'react-native-onyx';
+import Button from '@components/Button';
 import HeaderWithBackButton from '@components/HeaderWithBackButton';
 import ScreenWrapper from '@components/ScreenWrapper';
 import TabSelector from '@components/TabSelector/TabSelector';
+import Text from '@components/Text';
 import useLocalize from '@hooks/useLocalize';
+import useStyleUtils from '@hooks/useStyleUtils';
+import useTheme from '@hooks/useTheme';
 import useThemeStyles from '@hooks/useThemeStyles';
 import DebugUtils from '@libs/DebugUtils';
 import * as DeviceCapabilities from '@libs/DeviceCapabilities';
 import Navigation from '@libs/Navigation/Navigation';
 import OnyxTabNavigator, {TopTab} from '@libs/Navigation/OnyxTabNavigator';
 import type {DebugParamList} from '@libs/Navigation/types';
+import {getAllReportErrors} from '@libs/OptionsListUtils';
 import DebugDetails from '@pages/Debug/DebugDetails';
 import DebugJSON from '@pages/Debug/DebugJSON';
 import Debug from '@userActions/Debug';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
 import type SCREENS from '@src/SCREENS';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
 import DebugReportActions from './DebugReportActions';
 
 type DebugReportPageProps = StackScreenProps<DebugParamList, typeof SCREENS.DEBUG.REPORT>;
 
+type Metadata = {
+    title: string;
+    subtitle: string;
+    message?: string;
+    action?: {
+        name: string;
+        callback: () => void;
+    };
+};
+
 function DebugReportPage({
     route: {
         params: {reportID},
     },
 }: DebugReportPageProps) {
     const {translate} = useLocalize();
     const styles = useThemeStyles();
+    const StyleUtils = useStyleUtils();
+    const theme = useTheme();
     const [report] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`);
+    const [reportActions] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`);
+
+    const metadata = useMemo<Metadata[]>(() => {
+        if (!report) {
+            return [];
+        }
+
+        const reasonLHN = DebugUtils.getReasonForShowingRowInLHN(report);
+        const reasonGBR = DebugUtils.getReasonForShowingGreenDotInLHNRow(report);
+        const reportActionGBR = DebugUtils.getGBRReportAction(report, reportActions);
+        const reportActionRBR = DebugUtils.getRBRReportAction(report, reportActions);
+
+        return [
+            {
+                title: translate('debug.visibleInLHN'),
+                subtitle: translate(`debug.${!!reasonLHN}`),
+                message: reasonLHN ? translate(reasonLHN) : undefined,
+            },
+            {
+                title: translate('debug.GBR'),
+                subtitle: translate(`debug.${!!reasonGBR}`),
```

Comment:
Let's use the same condition in LHN

https://github.com/Expensify/App/blob/86360745431edb355c91e5594c741a54e5f88581/src/components/LHNOptionsList/OptionRowLHN.tsx#L117

Reply:
The only difference between the current implementation and what is in LHN is the `!hasBrickError`, as you can see below:

<img width="1064" alt="Screenshot 2024-10-01 at 16 32 31" src="https://github.com/user-attachments/assets/7eee9d82-d7d9-445c-b604-0517418ef5ad">

I will use `!reasonRBR` instead of `!hasBrickError`

Reply:
@pac-guerreiro The reason why I suggest using the same way is maybe in the future we can adjust the logic to display GBR (Ex: add some more cases), If we use the same method in two places, we don't need to worry about inconsistency in the future.

With the current change, in the future, if we adjust the logic in LHN, we easily miss an update in getReasonForShowingGreenDotInLHNRow

Reply:
@pac-guerreiro 

Let's see this case, you remove `isArchivedRoom` check block in `getReasonForShowingGreenDotInLHNRow`. I can cause the inconsistency. 
- The requiresAttentionFromCurrentUser function will early return if `isArchivedRoom` check block returns true
- The getReasonForShowingGreenDotInLHNRow function will continue to run in the below block

<img width="1366" alt="Screenshot 2024-10-02 at 10 25 19" src="https://github.com/user-attachments/assets/cd385d10-0366-4b4b-b47f-debcb59d5f68">

Reply:
@pac-guerreiro I see that we have a reason why we need to new function we need to return the reason instead of only detecting true or false. But It will be perfect if we have any ways to make the logic consistent all the time even if we have any new changes in the future. @pac-guerreiro Wdyt?